sunkit_image/coalignment/interface.py

```diff
@@ -0,0 +1,148 @@
+        have attributes for translation (dx, dy), scale, and rotation.
+
+    Returns
+    -------
+    `sunpy.map.Map`
+        A new sunpy map object with updated metadata reflecting the affine transformation.
+    """
+    pc_matrix = target_map.rotation_matrix
+    # Extacting the affine parameters
+    translation = affine_params.translation
+    scale = affine_params.scale
+    rotation_matrix = affine_params.rotation_matrix
+    # Updating the PC matrix
+    new_pc_matrix = pc_matrix @ rotation_matrix
+    reference_coord = reference_map.pixel_to_world(translation[0], translation[1])
+    Txshift = reference_coord.Tx - target_map.reference_coordinate.Tx
+    Tyshift = reference_coord.Ty - target_map.reference_coordinate.Ty
+    # Create a new map with the updated metadata
+    fixed_map = target_map.shift_reference_coord(Txshift, Tyshift)
+
+    fixed_map.meta["PC1_1"] = new_pc_matrix[0, 0]
+    fixed_map.meta["PC1_2"] = new_pc_matrix[0, 1]
+    fixed_map.meta["PC2_1"] = new_pc_matrix[1, 0]
+    fixed_map.meta["PC2_2"] = new_pc_matrix[1, 1]
+    
+    fixed_map.meta['cdelt1'] = (target_map.scale[0] / scale[0, 0]).value
+    fixed_map.meta['cdelt2'] = (target_map.scale[1] / scale[1, 1]).value
+
+    return fixed_map
+
+
+def warn_user_of_separation(reference_map,target_map):
+    """
+    Issues a warning if the separation between the reference and target maps is
+    large.
+
+    Parameters
+    ----------
+    reference_map : sunpy.map.Map
+        The reference map to which the target map is to be coaligned.
+    target_map : sunpy.map.Map
+        The target map to be coaligned to the reference map.
+    """
+    ref_coord = SkyCoord(reference_map.observer_coordinate)
+    target_coord = SkyCoord(target_map.observer_coordinate)
+    angular_separation = ref_coord.separation(target_coord)
+    if angular_separation > (1*u.deg):
+        warnings.warn(## warn due to large angular separation
+            "The angular separation between the reference and target maps is large. "
+            "This could cause errors when calculating shift between two "
+            "images. Please make sure the maps are close in space.",
+            SunpyUserWarning,
+            stacklevel=3,
+        )
+    # Calculate time difference and convert to separation angle
+    ref_time = reference_map.date
+    target_time = target_map.date
+    time_diff = abs(ref_time - target_time)
+    time_separation_angle = differential_rotation(time_diff.to(u.day), reference_map.center.Tx, model='howard')
+    if time_separation_angle > (1*u.deg):
+        warnings.warn(## warn due to large time separation
+            "The time difference between the reference and target maps in time is large. "
+            "This could cause errors when calculating shift between two "
+            "images. Please make sure the maps are close in time.",
+            SunpyUserWarning,
+            stacklevel=3,
+        )
+
+
+def coalign(reference_map, target_map, method):
```

Comment:
```suggestion
def coalign(reference_map, target_map, method='match_template'):
```

I think we should have a sensible default here, especially since this has been the only option for a long time anyway. I suspect this is the one most people are after.